api/src/bigquerytest/java/org/pmiops/workbench/cohorts/CohortMaterializationServiceTest.java

```diff
@@ -550,6 +573,45 @@ public void testMaterializeCohortPersonFieldSetPersonIdWithStringIsNull() {
     assertThat(response.getNextPageToken()).isNull();
   }
 
+  */
+
+  @Test
+  public void testMaterializeCohortPersonFieldSetPersonIdWithStringNotEqualNullNonMatch() {
+    TableQuery tableQuery = new TableQuery();
+    tableQuery.setTableName("person");
+    tableQuery.setColumns(ImmutableList.of("person_id"));
+    ColumnFilter filter = new ColumnFilter();
+    filter.setColumnName("ethnicity_source_value");
+    filter.setOperator(Operator.NOT_EQUAL);
+    filter.setValue("esv");
+    tableQuery.addFiltersItem(ImmutableList.of(filter));
+    FieldSet fieldSet = new FieldSet();
+    fieldSet.setTableQuery(tableQuery);
+    MaterializeCohortResponse response = cohortMaterializationService.materializeCohort(null,
+        SearchRequests.allGenders(), makeRequest(fieldSet, 1000));
+    assertPersonIds(response);
+    assertThat(response.getNextPageToken()).isNull();
+  }
+
+  @Test
+  public void testMaterializeCohortPersonFieldSetPersonIdWithStringIsNotNull() {
+    TableQuery tableQuery = new TableQuery();
+    tableQuery.setTableName("person");
+    tableQuery.setColumns(ImmutableList.of("person_id"));
+    ColumnFilter filter = new ColumnFilter();
+    filter.setColumnName("ethnicity_source_value");
+    filter.setOperator(Operator.NOT_EQUAL);
+    filter.setValueNull(true);
+    tableQuery.addFiltersItem(ImmutableList.of(filter));
+    FieldSet fieldSet = new FieldSet();
+    fieldSet.setTableQuery(tableQuery);
```

Comment:
So this doesn't have to do with the filters but why do we need the FieldSet? Currently it's just a wrapper for TableQuery that doesn't add any extra value. Could we just pass TableQuery instead or am i missing some added functionality that will appear later?

Reply:
In future I'm expecting we're going to have other types of queries under FieldSet (e.g. for fetching annotations, or particular views of the data that don't correspond to OMOP tables.) So having FieldSet at the top level allows us to add other children under it.